hal/common/file.hpp

```diff
@@ -65,7 +65,7 @@ class File
 	 * 
 	 * \return  success
 	 */
-	virtual bool open(const char* path) = 0;
+	virtual bool open(const char* path, bool new_file) = 0;
```

Comment:
For flash, we use seek to go back to start, for data logging, we just keep on writing at the end of the file. In both cases, the files will be open only once.

 In the case of several files in data logging, I do not see the problem: when you are finished with a file called filename.txt, call file.close(), then open a new file with file.open("newfilename.txt").
I do not see why you would like to call the open function but not open a file. The only case I see is if you try to open a new file although one is already open, the function file.open should return false and keep the previous file open.